src/state/swap/hooks.ts

```diff
@@ -232,16 +232,18 @@ export function useDerivedSwapInfo(platformOverride?: RoutablePlatform): UseDeri
     inputError = 'Insufficient ' + amountIn.currency.symbol + ' balance'
   }
 
-  if (useTradeExactInAllPlatformsRes.loading || useTradeExactOutAllPlatformsRes.loading) {
-    dispatch(setLoading(true))
-  } else {
-    dispatch(setLoading(false))
-  }
-  if (inputError !== undefined) {
-    setTimeout(() => {
+  useLayoutEffect(() => {
+    if (inLoading || outLoading) {
+      dispatch(setLoading(true))
+    } else {
       dispatch(setLoading(false))
-    }, 500)
-  }
+    }
+    if (inputError !== undefined) {
+      setTimeout(() => {
+        dispatch(setLoading(false))
+      }, 500)
+    }
+  }, [inLoading, outLoading, inputError, dispatch, typedValue, inputCurrency, outputCurrency])
```

Comment:
Does this fix #1055?

Reply:
@adamazad It was a bug earlier when I rebased the new loading state. It was triggering loading false in all the renders.

Not sure if it fix the above scenario.  @MilanVojnovic95 can you please check. I am not sure about all the scenarios.

Reply:
@wixzi  @adamazad 
Tested #1055 and it looks like it is fixed 👍 
I tested according to the same scenario and everything happens as expected.

Reply:
Thanks for the feedback @MilanVojnovic95!